### PR TITLE
Add cogatlas categories

### PR DESCRIPTION
**Pull request by RaphaelMeudec**
This PR contains:

- the current cognitive atlas task to category mapping (as json file)
- the script to generate the json file
- a modification to the figure generation to include the categories

### CODE REVIEW

**Comment by RaphaelMeudec** [comment]
@bthirion PR relative to Cogatlas categories

File: scripts/cogatlas_concepts_category_mapping.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+import json
+import requests
+from functools import reduce
+from pathlib import Path
+
+import pandas as pd
+import tqdm
+from bs4 import BeautifulSoup
+
+#%%
```
Comment (bthirion) [inline scripts/cogatlas_concepts_category_mapping.py]: you can remove that one :-)

File: scripts/cogatlas_concepts_category_mapping.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+def parse_mappings_from_categories_page():
+    response = requests.get("https://www.cognitiveatlas.org/concepts/categories/all")
```
Comment (bthirion) [inline scripts/cogatlas_concepts_category_mapping.py:15]: A 1-line doctring ?

File: scripts/cogatlas_concepts_category_mapping.py [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+            mapping_concept_to_category[concept_name] = category_name
+            mapping_category_to_concepts[category_name] = concept_name
+
+    return mapping_concept_to_category, mapping_category_to_concepts
+
+mapping_concept_to_category, mapping_category_to_concepts = parse_mappings_from_categories_page()
+
+#%%
+def get_cogatlas_concepts():
+    url = "https://www.cognitiveatlas.org/api/v-alpha/concept?format=json"
```
Comment (bthirion) [inline scripts/cogatlas_concepts_category_mapping.py:40]: same thing

File: scripts/cogatlas_concepts_category_mapping.py [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+)
+
+#%%
+missing_labels_in_mapping = [
+    key for key in unique_labels
+    if key not in mapping_concept_to_category.keys()
+]
+
+#%%
+def find_cogatlas_concept_by_name(name, cogatlas_concepts):
```
Comment (bthirion) [inline scripts/cogatlas_concepts_category_mapping.py:60]: Same thing.

**Review by bthirion — COMMENTED** [review]
LGTM. Just a bit of cleaning...

**Review by bthirion — APPROVED** [review]
LGTM.
We should consider adding a layer of tests for the core functions.

**Pull request merged**